proposals/ref-readonly-parameters.md

```diff
@@ -0,0 +1,112 @@
+# `ref readonly` parameters
+
+## Summary
+[summary]: #summary
+
+Allow parameter declaration-site modifier `ref readonly` and change callsite rules as follows:
```

Comment:
trying to wrap my head around this. I though `in` was literally our shorthand for `ref readonly`.  We even have that in our own code:

![image](https://github.com/dotnet/csharplang/assets/4564579/3f14dccf-f9b7-4350-a778-609b5f5a6c07)

Are we consdiring making it so that `in` vs `ref readonly` on a parameter would have different semantics?

Reply:
You probably need to add a `RefReadOnlyInput = 4`  or something. The `RefReadOnly` one at least specifics in the comments that it is only for returns, let's hope people read that :)